src/Xrefcheck/Core.hs

```diff
@@ -363,11 +376,15 @@ data VerifyProgress = VerifyProgress
 
 initVerifyProgress :: [Reference] -> VerifyProgress
 initVerifyProgress references = VerifyProgress
-  { vrLocal = initProgress (length localRefs)
-  , vrExternal = initProgress (length (ordNub $ map rLink extRefs))
+  { vrLocal = initProgress . length . filter isNothing $ extLinks
+  , vrExternal = initProgress . length . ordNub . catMaybes $ extLinks
   }
   where
-    (extRefs, localRefs) = L.partition (isExternal . rInfo) references
+    extLinks = flip fmap references $ \ref ->
+      case rInfo ref of
+        RIExternal (ELUrl url) ->
+          Just url
+        _ -> Nothing
```

Comment:
A challenge: will you be able to write the same using prisms from `lens` package?

You can autogenerate prisms using `makePrisms`, and then use those lens to filter differently.

Also, I suppose `ELOther` should not be included into "local" part (in all fairness, previously it was included into "external" part which seems to be also invalid). Wonder if #235 it would be possible (and viable) to force the initialization of progress bar to be also not so error-prone.

Reply:
Cool, I will try to do so!

> previously it was included into "external" part

I think that previously it was also included in the local part. `isExternal` `RIOtherProtocol` was defined to be `False`.

Reply:
I have modified this to not include `ELOther` in the progress bar, which also required to modify `modifyProgressRef` because it was also updating progress corresponding to `ELOther` references in the local progress bar.

Reply:
> Wonder if https://github.com/serokell/xrefcheck/issues/235 it would be possible (and viable) to force the initialization of progress bar to be also not so error-prone.

Regarding this, I have added an initialization function that accepts a list of "progress unit witnesses", so it is currently a list of `()` for the local part and a list of `Text` for the external one.